src/persistence/settings.cpp

```diff
@@ -838,6 +852,72 @@ void Settings::setTranslation(QString newValue)
     translation = newValue;
 }
 
+void Settings::deleteToxme()
+{
+    setToxmeInfo("");
+    setToxmeBio("");
+    setToxmePriv("");
+    setToxmePass("");
+}
+
+void Settings::setToxme(QString name, QString server, QString bio, bool priv, QString pass)
+{
+    setToxmeInfo(name + "@" + server);
+    setToxmeBio(bio);
+    setToxmePriv(priv);
+    if (!pass.isEmpty())
+        setToxmePass(pass);
+}
+
+QString Settings::getToxmeInfo() const
+{
+    QMutexLocker locker{&bigLock};
+    return toxmeInfo;
+}
+
+void Settings::setToxmeInfo(QString info)
+{
+    QMutexLocker locker{&bigLock};
+    if (info.split("@").size() == 2)
```

Comment:
It is service limit.
I may replace split on lastIndexOf, but what for? It will still incorrect name :)